MapboxCoreNavigation/RouteController.swift

```diff
@@ -162,7 +164,48 @@ extension RouteController: CLLocationManagerDelegate {
         let radius = min(RouteControllerMaximumDistanceBeforeRecalculating,
                          location.horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
 
-        return newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
+        let isCloseToCurrentStep = newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
+        
+        // If the user is moving away from the maneuver location
+        // and they are close to the next step
+        // we can safely say they have completed the maneuver.
+        // This is intended to be a fallback case when we do find
+        // that the users course matches the exit bearing.
+        if let upComingStep = routeProgress.currentLegProgress.upComingStep {
+            let isCloseToUpComingStep = newLocation.isWithin(radius, of: upComingStep)
+            if !isCloseToCurrentStep && isCloseToUpComingStep && isInsideManeuverRadius {
```

Comment:
Why does this behavior depend on being within the maneuver zone? If somehow the user has made it to the next step without a location update coming in as they pass through the maneuver zone, we probably would want to advance to the next step regardless of the maneuver zone. You could imagine this case becoming relevant for roundabouts and large intersections.